src/charset.c

```diff
@@ -840,12 +852,22 @@ linetabsize_no_outer(win_T *wp, linenr_T lnum)
     void
 win_linetabsize_cts(chartabsize_T *cts, colnr_T len)
 {
+    vimlong_T vcol = cts->cts_vcol;
 #ifdef FEAT_PROP_POPUP
     cts->cts_with_trailing = len == MAXCOL;
 #endif
     for ( ; *cts->cts_ptr != NUL && (len == MAXCOL || cts->cts_ptr < cts->cts_line + len);
 						      MB_PTR_ADV(cts->cts_ptr))
-	cts->cts_vcol += win_lbr_chartabsize(cts, NULL);
+    {
+	vcol += win_lbr_chartabsize(cts, NULL);
+	if (vcol > MAXCOL)
+	{
+	    cts->cts_vcol = MAXCOL;
+	    break;
+	}
+	else
+	    cts->cts_vcol = (int)vcol;
+    }
 #ifdef FEAT_PROP_POPUP
     // check for a virtual text at the end of a line or on an empty line
     if (len == MAXCOL && cts->cts_has_prop_with_text && *cts->cts_ptr == NUL)
```

Comment:
Is overflow check for the `cts_vcol += below needed as well?

Reply:
```diff
diff --git a/src/charset.c b/src/charset.c
index d9ac6889a..544c67575 100644
--- a/src/charset.c
+++ b/src/charset.c
@@ -863,7 +863,7 @@ win_linetabsize_cts(chartabsize_T *cts, colnr_T len)
 	if (vcol > MAXCOL)
 	{
 	    cts->cts_vcol = MAXCOL;
-	    break;
+	    return;
 	}
 	else
 	    cts->cts_vcol = (int)vcol;
@@ -873,11 +873,12 @@ win_linetabsize_cts(chartabsize_T *cts, colnr_T len)
     if (len == MAXCOL && cts->cts_has_prop_with_text && *cts->cts_ptr == NUL)
     {
 	(void)win_lbr_chartabsize(cts, NULL);
-	cts->cts_vcol += cts->cts_cur_text_width;
+	vcol += cts->cts_cur_text_width;
 	// when properties are above or below the empty line must also be
 	// counted
 	if (cts->cts_ptr == cts->cts_line && cts->cts_prop_lines > 0)
-	    ++cts->cts_vcol;
+	    ++vcol;
+	cts->cts_vcol = vcol > MAXCOL ? MAXCOL : (int)vcol;
     }
 #endif
 }
```

Reply:
Yeah, I thought about it, but I assumed `cts->cts_ptr` would be non-NULL then (however that condition may not always be true). So it's probably better to add this as well. Thanks.